numba/parfor.py

```diff
@@ -245,31 +270,44 @@ def run(self):
         # e.g. convert A.sum() to np.sum(A) for easier match and optimization
         canonicalize_array_math(self.func_ir, self.typemap,
                                 self.calltypes, self.typingctx)
-        self._replace_parallel_functions(self.func_ir.blocks)
+        # some numpy functions are given parallel implementations
+        if self.options.numpy:
+            self._replace_parallel_functions(self.func_ir.blocks)
+        # run array analysis, a pre-requisite for parfor translation
```

Comment:
Currently `prange` is always on, and requires array analysis. 

Initially I thought `prange` is a more fine-grained control over these options since it can be individually changed to `range`. But now I think it is useful to have `prange` off while leaving others on for debugging purpose. So I'm going to add a `prange` option too.

For array analysis, it is at least harmless to leave it even when all other options are turned off.